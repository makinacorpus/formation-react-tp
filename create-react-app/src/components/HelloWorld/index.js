import React from 'react';

class HelloWorld extends React.Component {
  render() {
    return (
      <div className="HelloWorld">Hello {this.props.name}</div>
    );
  }
}

export default HelloWorld;
