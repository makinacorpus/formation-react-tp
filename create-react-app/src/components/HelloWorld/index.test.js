import React from 'react';
import ReactDOM from 'react-dom';

import renderer from 'react-test-renderer';

import HelloWorld from './';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<HelloWorld />, div);
});

it('renders without crashing with attr name', () => {
  const div = document.createElement('div');
  ReactDOM.render(<HelloWorld name='world' />, div);
});

it('renders correctly to snapshot', () => {
  const tree = renderer.create(
    <HelloWorld />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly to snapshot', () => {
  const tree = renderer.create(
    <HelloWorld name='world'/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

