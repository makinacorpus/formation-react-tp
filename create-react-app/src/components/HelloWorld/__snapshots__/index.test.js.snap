// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly to snapshot 1`] = `
<div
  className="HelloWorld"
>
  Hello 
</div>
`;
